agent/pkg/server/server.go

```diff
@@ -425,14 +425,24 @@ func unmarshalConfigItemsJson(project string, items []*pb.ConfigItem) (map[strin
 			default:
 				// Structured value (object, array, number, boolean)
 				// Set both Value (JSON string) and ObjectValue (natural value)
-				// Use protojson to properly handle protobuf semantics
-				jsonBytes, err := protojson.Marshal(vv.Value)
+				// Use protojson to handle protobuf semantics, then normalize with json.Marshal
+				protoJsonBytes, err := protojson.Marshal(vv.Value)
 				if err != nil {
 					return nil, fmt.Errorf("marshaling %q value to JSON: %w", item.Key, err)
 				}
+				// Unmarshal to Go value to normalize the representation
+				var obj interface{}
+				if err := json.Unmarshal(protoJsonBytes, &obj); err != nil {
+					return nil, fmt.Errorf("unmarshaling %q protojson: %w", item.Key, err)
+				}
+				// Re-marshal with standard json.Marshal for deterministic compact output
+				jsonBytes, err := json.Marshal(obj)
+				if err != nil {
+					return nil, fmt.Errorf("re-marshaling %q value: %w", item.Key, err)
+				}
 				jsonStr := string(jsonBytes)
 				cv.Value = &jsonStr
-				cv.ObjectValue = val
+				cv.ObjectValue = obj
```

Comment:
I'm actually fixing two bugs here; it was incorrect to use `val` directly as `cv.ObjectValue`, because `cv` is later encoded using `json.Marshal`.  The correct approach is for `cv.ObjectValue` to have a pure Go value, as shown here.

I updated the tests to use a more deeply-nested object, to better verify the protojson goodness.